src/osd/scheduler/mClockScheduler.h

```diff
@@ -186,15 +206,35 @@ class mClockScheduler : public OpScheduler, md_config_obs_t {
   // Enqueue op in the back of the regular queue
   void enqueue(OpSchedulerItem &&item) final;
 
-  // Enqueue the op in the front of the regular queue
+  // Enqueue the op in the front of the high priority queue or the immediate queue (based on priority)
   void enqueue_front(OpSchedulerItem &&item) final;
 
+  // Enqueue the op to the high priority queue
+  void enqueue_strict(unsigned prio, OpSchedulerItem &&item, bool front = false);
+
   // Return an op to be dispatch
   WorkItem dequeue() final;
 
+  bool high_priority_empty() const {
+    if (high_priority.empty()) {
+      return true;
+    }
+    for (auto it = high_priority.begin();
+	 it != high_priority.end(); it++) {
+      if (!it->second.empty()) {
+	return false;
+      }
+    }
+    return true;
+  }
+
   // Returns if the queue is empty
   bool empty() const final {
-    return immediate.empty() && scheduler.empty();
+    if (immediate.empty() && scheduler.empty()
```

Comment:
`return immediate.empty() && scheduler.empty() && high_priority_empty();`

You could just do with the above in one line.